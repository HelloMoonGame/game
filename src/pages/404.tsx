import { Box, Container, Typography, Button } from '@material-ui/core'
import Head from 'next/head'
import { useRouter } from 'next/router'
import React from 'react'

const PageNotFound = (): JSX.Element => {
  const router = useRouter()

  return (
    <>
      <Head>
        <title>Page not found - Hello Moon</title>
      </Head>
      <Box
        display="flex"
        flexDirection="column"
        height="100%"
        justifyContent="center"
      >
        <Container maxWidth="md">
          <Typography align="center" color="textPrimary" variant="h2">
            Page not found
          </Typography>
          <Typography align="center" color="textPrimary" variant="h6">
            Sorry, the page you are looking for does not exist
          </Typography>
          <Box textAlign="center" marginTop="20px">
            <Button
              variant="contained"
              color="primary"
              onClick={() => router.push('/')}
            >
              Homepage
            </Button>
          </Box>
        </Container>
      </Box>
    </>
  )
}

export default PageNotFound
