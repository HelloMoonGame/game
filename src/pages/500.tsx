import { Box, Container, Typography, Button } from '@material-ui/core'
import Head from 'next/head'
import { useRouter } from 'next/router'
import React from 'react'

const ErrorPage = (): JSX.Element => {
  const router = useRouter()

  return (
    <>
      <Head>
        <title>Error - Hello Moon</title>
      </Head>
      <Box
        display="flex"
        flexDirection="column"
        height="100%"
        justifyContent="center"
      >
        <Container maxWidth="md">
          <Typography align="center" color="textPrimary" variant="h2">
            Error
          </Typography>
          <Typography align="center" color="textPrimary" variant="h6">
            Sorry, something went wrong!
          </Typography>
          <Box textAlign="center" marginTop="20px">
            <Button
              variant="contained"
              color="primary"
              onClick={() => router.push('/')}
            >
              Homepage
            </Button>
          </Box>
        </Container>
      </Box>
    </>
  )
}

export default ErrorPage
