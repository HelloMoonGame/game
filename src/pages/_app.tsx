import React from 'react'
import { AppProps } from 'next/app'
import { createMuiTheme, CssBaseline, ThemeProvider } from '@material-ui/core'
import Layout from '../components/Layout'
import { Provider } from 'react-redux'

import store from '../state/store'
import AuthenticationCheck from '../components/AuthenticationCheck'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#0b3663',
    },
    secondary: {
      main: '#fdd300',
    },
  },
})

export default function MyApp({ Component, pageProps }: AppProps): JSX.Element {
  return (
    <Provider store={store}>
      <CssBaseline />
      <AuthenticationCheck>
        <ThemeProvider theme={theme}>
          <Layout>
            <Component {...pageProps} />
          </Layout>
        </ThemeProvider>
      </AuthenticationCheck>
    </Provider>
  )
}
