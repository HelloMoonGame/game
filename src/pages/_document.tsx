import React from 'react'
import Document, { Html, Head, Main, NextScript } from 'next/document'

export default class MyDocument extends Document {
  render(): JSX.Element {
    return (
      <Html lang="en">
        <Head />
        <body>
          <style jsx global>{`
            html,
            body {
              height: 100%;
            }
            #__next {
              display: flex;
              height: 100%;
            }
          `}</style>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}
