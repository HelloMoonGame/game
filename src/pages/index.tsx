import React from 'react'
import GameMap from '../components/gamemap/GameMap'

export const Home = (): JSX.Element => <GameMap />

export default Home
