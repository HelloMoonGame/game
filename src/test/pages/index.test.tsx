import React from 'react'
import client, { Session } from 'next-auth/client'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

jest.mock('next-auth/client')

describe('Home page', () => {
  it('matches snapshot', () => {
    const mockSession: Session = {
      expires: '1',
      user: { email: 'a', name: 'Delta', image: 'c' },
    }

    ;(client.useSession as jest.Mock).mockReturnValueOnce([mockSession, false])

    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
