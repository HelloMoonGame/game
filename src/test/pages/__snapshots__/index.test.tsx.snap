// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Hello Moon
        </h6>
      </div>
    </header>
  </header>
  <main
    class="makeStyles-pageMain-1"
  >
    <canvas
      class="makeStyles-canvas-3"
      height="0"
      width="0"
    />
    <canvas
      class="makeStyles-canvas-3"
      height="0"
      width="0"
    />
    <div
      class="makeStyles-canvas-3"
    />
  </main>
  <footer
    class="makeStyles-pageFooter-2"
  >
    <div
      class="MuiBottomNavigation-root"
    >
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiBottomNavigationAction-wrapper"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            location_on
          </span>
          <span
            class="MuiBottomNavigationAction-label"
          >
            Explore
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </footer>
</DocumentFragment>
`;
